src/fairseq2/recipes/hg/asr_eval.py

```diff
@@ -82,97 +125,133 @@ class AsrEvalConfig:
     """The data type of the model."""
 
 
-asr_eval_presets = ConfigRegistry[AsrEvalConfig]()
+@dataclass
+class EvalSeqBatch:
```

Comment:
Any reason for not using `Seq2SeqBatch` [here](https://github.com/facebookresearch/fairseq2/blob/main/src/fairseq2/models/seq2seq.py#L53) like the native ASR evaluator [here](https://github.com/facebookresearch/fairseq2/blob/main/src/fairseq2/recipes/wav2vec2/asr/eval.py#L232)

Reply:
The `Seq2SeqBatch` class expects targets to be tensors, which is useful for training. However, during evaluation, encoding the target only to decode it again without utilizing the tensors seems redundant.

Is there a better approach? another fairseq2 data-structure to use?